GoMeet/src/main/webapp/test/spec/spec-meeting-event-dao.js

```diff
@@ -0,0 +1,208 @@
+/**
+ * Creates <input> elements with the names and values as given in arr.
+ * @param {Array} arr Array of objects each in the format {name: value},
+ * representing the name and value of the <input> elements to create.
+ * @returns {Array} Array of <input> elements created
+ */
+function createInputElems(arr) {
+  let inputElements = [];
+  for (let i = 0; i < arr.length; i++) {
+    let elem = document.createElement('input');
+    let name = Object.keys(arr[i])[0];
+    let value = arr[i][name];
+    elem.name = name;
+    elem.value = value;
+    inputElements.push(elem);
+  }
+  return inputElements;
+}
+
+describe ('MeetingEventDAO - getter functions', function() {
+  // Set up
+  let documentSpy;
+  let fakeSessionStorage = {};
+  const MEETING_NAME = 'Christmas Lunch';
+  const DURATION_MINS = '30'; 
+  const DURATION_HOURS = '1'; 
+  const TIME_FIND_METHOD = 'manual';
+  const HOST = 'vivian@gmail.com';
+  const GUEST_1 = 'anna@gmail.com';
+  const GUEST_2 = 'therese@gmail.com';
+  const TIME_1 = '2021-02-02T08:19';
+  const TIME_2 = '2021-02-05T08:19';
+  const FAKE_KEYS = ['meeting-name', 'meeting-host', 'guest-1', 'guest-3', 'duration-mins', 
+      'duration-hours', 'time-find-method', 'meeting-time-1', 'meeting-time-4'];
+
+  beforeAll(function () {
+    // Hardcode some data to store to session storage
+    // Note: getGuestList() should return all sessionStorage entries with the 
+    // prefix 'guest-' regardless of the number. Same goes for getMeetingTimes()
+    let inputElemData = [
+      { 'meeting-name': MEETING_NAME },
+      { 'meeting-host': HOST },
+      { 'guest-1': GUEST_1 }, 
+      { 'guest-3': GUEST_2 }, 
+      { 'duration-mins': DURATION_MINS }, 
+      { 'duration-hours': DURATION_HOURS }, 
+      { 'time-find-method': TIME_FIND_METHOD }, 
+      { 'meeting-time-1': TIME_1 }, 
+      { 'meeting-time-4': TIME_2 }
+    ];
+    
+    let inputElements = createInputElems(inputElemData);
+    inputElements = new MockHTMLCollection(inputElements);
```

Comment:
just a comment: I'm not sure it's generally great style to be switching around datatypes for a variable.

Reply:
my bad!